internal/github/github.go

```diff
@@ -95,9 +129,15 @@ func AuthenticateWithToken(ctx context.Context, token string, pendingRecheckTime
 	)
 	ctx = context.WithValue(ctx, oauth2.HTTPClient, &http.Client{Transport: cachingRetryableTransport()})
 	httpClient := oauth2.NewClient(ctx, src)
-	githubClient := github.NewClient(httpClient)
 
-	return GHClient{client: githubClient, pendingRecheckTime: pendingRecheckTime}
+	restClient := github.NewClient(httpClient)
+	graphQLClient := graphql.NewClient("https://api.github.com/graphql", httpClient)
```

Comment:
I noticed this client isn't using the proper logger ...

```console
INFO[2025-01-23 18:17:40] Checking CI status on iainlane/test@7456abc7104e4138951a72b6ca0dc0039d58ed62 
2025/01/23 18:17:40 [DEBUG] POST https://api.github.com/graphql
CI successful
```

See the two styles of message?

We're using logrus in `wait-for-github` but that doesn't plug straight into `go-retryablehttp`'s `Logger` field. So I think we should fix this separately, maybe by switching to `slog`, which does work fine.

I think it's more appropriate to do that in a follow-up PR, as it's a separate issue.